client/src/App.tsx

```diff
@@ -24,21 +25,23 @@ function App(): JSX.Element {
         <SnackBarProvider>
           <AuthProvider>
             <SocketProvider>
-              <Switch>
-                <Route exact path="/login" component={Login} />
-                <Route exact path="/signup" component={Signup} />
-                <ProtectedRoute exact path="/dashboard" component={Dashboard} />
-                <Route exact path="/listings" component={Listings} />
-                <Route path="/listings/:sitterId" component={ProfileDetail} />
-                <Route exact path="/profile" component={Profile} />
-                <Route path="/profile/:setting" component={Profile} />
-                <Route exact path="/messages" component={ChatSideBanner} />
-                <Route exact path="/my-sitters" component={MySitters} />
-                <Route exact path="/my-jobs" component={MyJobs} />
-                <Route path="*">
-                  <Redirect to="/login" />
-                </Route>
-              </Switch>
+              <ConvoProvider>
```

Comment:
It would be better to wrap only the 'Messages' component with the `ConvoProvider` as each time we receive a new message, all of the children within this provider are re-rendered :)

Reply:
Applied